src/git.rs

```diff
@@ -22,15 +27,41 @@ pub fn repo_hash<P: AsRef<Path>>(path: P) -> Option<String> {
     }
 }
 
-fn dirty<P: AsRef<Path>>(path: P) -> bool {
-    let path = path.as_ref().to_str();
-    let mut args = match path {
-        Some(path) => vec!["-C", path],
-        None => vec![],
-    };
-    args.extend(&["ls-files", "-m"]);
-    match Command::new("git").args(&args).output() {
+fn dirty(path: impl AsRef<Path>) -> bool {
+    match git_command(path.as_ref(), &["ls-files", "-m"]) {
         Ok(modified_files) => !modified_files.stdout.is_empty(),
         Err(_) => false,
     }
 }
+
+fn gitmodules_config(subcommand: &[&str]) -> Option<String> {
+    let mut args = vec!["config", "-f", ".gitmodules", "-z"];
+    args.extend(subcommand);
+    let output = git_command(Path::new("."), &args).ok()?;
```

Comment:
Perhaps should just use `Command::new` here directly. Saves an extra `vec![]` heap allocation + copying over values from the slice, and unnecessary `["-C", "."]` argument.

Reply:
But probably is more code? I don't think performance concerns are valid here for this code. Write it simple and understandable, not fast :)

Reply:
Maybe it should also take an `Option` for the path?

Reply:
> But probably is more code?

Fortunately not really, this is all we need from `git_command`:

```suggestion
    let output = Command::new("git").args(&args).output().ok()?;
```

Not so much about performance, just seems cleaner that way. `git_command` is strangely using an `Option` already to hide when the path is not a valid UTF-8 string, rather than panicking :raised_eyebrow:

Reply:
That seems broken (the `Option` thing). It should rather panic in that case...

Reply:
Indeed. Fixed it as part of this PR, just in case.